csv-bulk-signature/src/main.ts

```diff
@@ -201,11 +180,37 @@ async function getCSVRows(path: string) {
   });
 }
 
+/* filter out the first row
+ * & filter out incomplete contacts
+ */
+async function filterCompleteContacts(contacts: Contact[]) {
+  const filteredContacts = contacts.filter((contact) => {
+    // @ts-ignore - a hack to skip the first row
+    return contact["Brand*"] !== "use dropdown to select a brand";
+  });
+
+  // check if all required fields are present and log out skipped rows
+  const contactsWithRequiredFields = filteredContacts.filter((contact) => {
+    const hasRequiredData = checkRequiredFields(contact);
+    if (!hasRequiredData) {
+      skippedRows.push(contact["Full Name*"]);
+    } else {
+      processedRows.push(contact["Full Name*"]);
+    }
+    return hasRequiredData;
+  });
+
+  return contactsWithRequiredFields;
+}
+
 async function main() {
   await setupFolders();
   const template = await getTemplate();
-  await checkHeadersToBeSame();
-  await generateSignatures(template);
+
+  const contacts = await getCSVRows(CSV_FILE);
+  await checkHeadersToBeSame(contacts[0]);
+  const completeContacts = await filterCompleteContacts(contacts);
+  await generateSignatures(template, completeContacts);
```

Comment:
get CSV data -> check first contact for headers -> filter contacts -> generate signatures